f8a_report/helpers/npm_metadata.py

```diff
@@ -0,0 +1,232 @@
+# Copyright © 2020 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Author: Dharmendra G Patel <dhpatel@redhat.com>
+#
+"""Class to update NPM package details."""
+import datetime
+import requests
+import logging
+from helpers.s3_helper import S3Helper
+
+logger = logging.getLogger(__file__)
+
+NPM_PACKAGE_FILE_PATH = "training-utils/node-package-details.json"
+
+
+class NPMMetadata:
+    """NPM metadata fetcher."""
+
+    def __init__(self, s3Helper=S3Helper, github_token=str,
+                 bucket_name=str, manifest_data=dict()):
+        """Set obect default memebers."""
+        self.s3Helper = s3Helper
+        self.github_token = github_token
+        self.bucket_name = bucket_name
+        self.package_list = self._flatten_list(manifest_data)
+        self.existing_data = self._load_data()
+
+        self.stats = {
+            'existing_count': len(self.existing_data),
+            'unique_manifest_count': len(self.package_list),
+            'metadata_exists': 0,
+            'total_missing': 0,
+            'new_packages': 0,
+            'updated_count': 0,
+            'still_missing': 0,
+            'fetched_from_npm': 0,
+            'fetched_from_github': 0,
+            'npm_fetch_errors': 0,
+            'github_fetch_errors': 0,
+        }
+
+    def update(self):
+        """Read and update metadata for all NPM packages in S3."""
+        logger.info("Existing node package length: %d", self.stats['existing_count'])
```

Comment:
Can you reduce the number of info logs? It begins to affect readability of logs when you need to debug an issue.

Reply:
This one is printed once for a trigger, so I would like to keep it. But agree to your point on reducing logs. I will remove some info logs which are printed in a loop.